starter-core/src/main/java/io/micronaut/starter/feature/graalvm/GraalVMFeatureValidator.java

```diff
@@ -39,7 +39,7 @@ public void validatePreProcessing(Options options, ApplicationType applicationTy
     @Override
     public void validatePostProcessing(Options options, ApplicationType applicationType, Set<Feature> features) {
         if (features.stream().anyMatch(f -> f instanceof GraalVM || f instanceof GraalVMDockerRegistryWorkflow)) {
-            if (options.getLanguage() == Language.GROOVY) {
+            if (!supports(options.getLanguage())) {
                 throw new IllegalArgumentException("GraalVM is not supported in Groovy applications");
             }
```

Comment:
Refactor to avoid having Groovy hardcoded in the condition is great. But then, the exception message is still hardcoded. This change is not worth it IMO

Reply:
> Refactor to avoid having Groovy hardcoded in the condition is great. But then, the exception message is still hardcoded

Good catch. I missed the harcoded string. I have removed it here: 

https://github.com/micronaut-projects/micronaut-starter/pull/1466/commits/42bda3299a2e074881e7b209c32d7b5da0ab420c